engine/classes/ElggEntity.php

```diff
@@ -1444,24 +1488,34 @@ public function isFullyLoaded() {
 	 * @throws IOException
 	 */
 	public function save() {
-		$guid = $this->getGUID();
-		if ($guid > 0) {
+		if ($this->is_saving) {
+			elgg_log("Do not save() entities while they are being saved.", 'WARNING');
```

Comment:
I can imagine this message being quite confusing for a novice. Could it be made more specific?